OpenSSL info retrofit into 1.4 (#143)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/enable-inbound-traffic.adoc

+
It can take up to 5 minutes for any permissions you gain to propagate.

.. From Anypoint Platform, select *Access Management*.
+

> If you do not have access to Access Management, request access from your system administrator.

Reviewer: ```suggestion
If you do not have it, request access to the access management service from your system administrator.
```

.. Select *Users*

> .. Locate the row with your username and then select the blue link.

Reviewer: ```suggestion
.. Locate the row that contains your username and select the blue link.
```

.. On the *Runtime Manager* tab, enable the *Manage Runtime Fabrics* permission for your Runtime Fabric environment.

.. On the *Secrets Manager* tab, enable the *Manage Secret Groups*, *Write Secrets*, and *Read Secrets Metadata* permissions for your Runtime Fabric environment.

== Generate a Certificate-Key Pair
You need a certificate-key pair for the internal load balancer in Runtime Fabric. If you have one already, both files must be of the .pem type, your keystore must have a passcode, and your common name should match the domain you use in your Runtime Fabric. 

RSA keys are the most common type of keys. RSA keys of 2K length offer the best compromise between security and performance. 
[NOTE]

> RSA keys larger than 2K protect against brute force cracking and are appropriate for certificates that have expirations of many years. However, whenever key length is doubled, for example, from 2k to 4k, performance is reduced by a factor greater than 6.

Reviewer: What do we use length as the unit of measure for 2K above, but size here? Seems inconsistent to me, and I commented on it last time as well.

 
ECDSA keys are also supported. In most cases, ECDSA doubles the performance of a 2K RSA key. Supported curves are:

* secp521r1 (P-521)
* secp384r1 (P-384)
* secp256r1 (also known as prime256v1 (P-256))

Use the following steps to generate a certificate-key pair for evaluation purposes.

. Run the following command on your machine to generate a certificate-key pair:
+
----
openssl req -x509 -newkey rsa:2048 -keyout key.pem -out cert.pem -days 365
----

. Type a passphrase for your key.

> . Complete the requested information. When asked for a common name, supply the domain to be used in your Runtime Fabric.

Reviewer: Please ensure that list items and steps are one sentence max.:
```suggestion
. Complete the requested information.
When asked for a common name, supply the domain to be used in your Runtime Fabric.
```

> If you use a wildcard, for example, `*.example.com` in your common name, your application URLs use the following format: `{app-name}.example.com`. Otherwise, your application URLs use the format `example.com/{app-name}`.

Reviewer: ```suggestion
If you use a wildcard, for example, `*.example.com`, in your common name, your application URLs use the following format: `{app-name}.example.com`. Otherwise, your application URLs use the format `example.com/{app-name}`.
```
I edited previously to use parentheses to provide relief from multiple commas in one sentence. That was obviously rejected, but you do need, as a result, this one more.

[NOTE]
Make sure you’re in the right business group and select your Runtime Fabric environment in the sidebar on the left.

> . Select the *Create Secret Group* button.

Reviewer: Buttons are clicked. Choices are selected. This standard is not going to change without further discussion, and your deliverables will be worded differently from the rest in the library.

. Under *General*, type the name of your secret group.
. Select *Save*.
. Select *Keystore* in the sidebar and create your keystore:
.. Select *Add Keystore*.
+

> If you don’t see the *Add Keystore* button, ensure that you are viewing your secret group in edit mode. Return to *Secret Groups* and then select the *Edit* next to your group.

Reviewer: ```suggestion
If you don’t see the *Add Keystore* button, ensure that you are viewing your secret group in edit mode. Return to *Secret Groups* and select *Edit* next to your group.
```

.. Enter a name for your keystore.
+

> This name identifies the keystore when you add it to your TLS context below.

Reviewer: ```suggestion
This name identifies the keystore when you add it to your TLS context later.
```

.. Under *Certificate File*, select *Choose File* and then select your `.pem` certificate` file.
.. Under *Key File*, select *Choose File* and then select your `.pem` key file.
.. Enter your key passphrase.

> .. Optionally, upload a separate certificate file containing a CA path.

Reviewer: Can we replace this with an "If..." statement that specifies WHY the user wants or needs to do this, rather than "optionally"?

.. Select *Save*.
+
[NOTE]

> If you receive an unknown error, it might be due to your Certificate file containing a CA path. You must move the CA path into its own file and upload it separately before continuing.

Reviewer: ```suggestion
If you receive an unknown error, it might be due to your certificate file containing a CA path. You must move the CA path into its own file and upload it separately before continuing.
```

. Select *TLS Context* and then perform the following:
+
[NOTE]

> If you do not see *Add TLS Context*, ensure you are viewing your secret group in edit mode. Return to the *Secret Groups* page and then select *Edit* next to your group.

Reviewer: This NOTE interrupts the flow implied by the previous colon and the subsequent step.
I edited for this previously.

Reviewer: ```suggestion
If you do not see *Add TLS Context*, ensure that you are viewing your secret group in edit mode. Return to the *Secret Groups* page and then select *Edit* next to your group.
```

> .. Type a name for your TLS context. This will identify it when you apply it to your Runtime Fabric later.

Reviewer: GLOBAL
```suggestion
.. Type a name for your TLS context.
This identifies it when you apply it to your Runtime Fabric later.
```

.. Use the default TLS version (TLS 1.2).

> .. Under *Target*, select *Anypoint Security*. This configures your TLS context so it can be applied to a Runtime Fabric.

Reviewer: GLOBAL
```suggestion
.. Under *Target*, select *Anypoint Security*. 
This configures your TLS context so it can be applied to a Runtime Fabric.
```

.. Under *Keystore*, select the name of the keystore you configured previously.
.. Select *Save*.

> . After saving your TLS context, select *Secret Groups* and then select *Finish*. This enables TLS Context for your *Runtime Fabric.*

Reviewer: GLOBAL
```suggestion
. After saving your TLS context, select *Secret Groups* and then select *Finish*. 
This enables TLS Context for your *Runtime Fabric.*
```

== Enable Inbound Traffic
The Runtime Fabric internal load balancer supports OpenSSL 1.1.1 and TLS 1.3, which provide the following security enhancements and performance improvements over TLS 1.2:
 
* 2x or greater TLS 1.2 connection performance throughput boost versus prior Runtime Fabric internal load balancer running on OpenSSL 1.0.2
* Reduction of one round trip in full handshake for TLS 1.3 vs. TLS 1.2
* Added TLS 1.2 and TLS 1.3 ChaCha20-Poly1305 ciphers for better mobile and IoT device support
* TLS 1.3 protection against downgrade attacks
* Continued support for both RSA and ECDSA keys

> * TLS 1.3 removes obsolete and insecure features from TLS 1.2 and TLS 1.1. Refer to the xref:release-notes::anypoint-security/anypoint-security-rtf-edge.adoc[Anypoint Security Edge Release Notes] for a complete list of supported and deprecated ciphers. 

Reviewer: I suggested making this parallel in my previous edit
```suggestion
* Removal of obsolete and insecure features from TLS 1.2 and TLS 1.1
Refer to the xref:release-notes::anypoint-security/anypoint-security-rtf-edge.adoc[Anypoint Security Edge Release Notes] for a complete list of supported and deprecated ciphers. 
```

> .. From the *TLS Context* menu, select the TLS Context to be used for your Runtime Fabric.

Reviewer: ```suggestion
.. From the *TLS Context* menu, select the TLS context to be used for your Runtime Fabric.
```

> . Select *Deploy* to deploy the internal load balancer. If successful, a message appears 

Reviewer: GLOBAL
```suggestion
. Select *Deploy* to deploy the internal load balancer. 
If successful, a message appears 
```

> in the bottom-right of the page. The deployment may take up to a minute. You can view the

Reviewer: ```suggestion
in the bottom-right of the page. The deployment might take up to a minute. You can view the
```
